### PR TITLE
chore(otoperator): update opentelemetry operator, add instrumentation customization

### PR DESCRIPTION
**Pull request by mat-rumian**
This PR:
* updates OpenTelemetry-Operator to latest `0.24.0`
* updates auto-instrumentation images used by OpenTelemetry-Operator
* add flags to disable/enable metrics and traces export from instrumentation
* increases default resources of OpenTelemetry-Operator pod to avoid OOM

### Checklist

- [x] Changelog updated or skip changelog label added
- [ ] Documentation updated
- [ ] Template tests added for new features
- [ ] Integration tests added or modified for major features
- [x] OpenTelemetry-Operator Python instrumentation test
- [x] OpenTelemetry-Operator NodeJS instrumentation test
- [x] OpenTelemetry-Operator Java instrumentation test
- [x] OpenTelemetry-Operator Dotnet instrumentation test
- [x] OpenTelemetry-Operator Instrumentation Resource creation test

### CODE REVIEW

File: deploy/helm/sumologic/values.yaml [hunk 1]
```diff
@@ -4517,6 +4539,13 @@ opentelemetry-operator:
 
   manager:
     env:
+    resources:
+      limits:
+        cpu: 250m
+        memory: 512Mi
+      requests:
+        cpu: 150m
+        memory: 256Mi
```
Comment (swiatekm-sumo) [inline deploy/helm/sumologic/values.yaml:4548]: Are these based on anything specific? I'd always like to keep requests as low as possible for services that don't really need to scale.

File: deploy/helm/sumologic/templates/opentelemetry-operator/job-instrumentation-cr.yaml [hunk 2]
```diff
@@ -6,12 +7,15 @@
 apiVersion: batch/v1
 kind: Job
 metadata:
-  name: {{ template "sumologic.metadata.name.opentelemetry.operator.instrumentation.job" . }}
+  name: {{ template "sumologic.metadata.name.opentelemetry.operator.instrumentation.job" . }}-{{ include (print $.Template.BasePath "/opentelemetry-operator/configmap-instrumentation-cr.yaml") $ctx | sha256sum | substr 0 5 }}
```
Comment (swiatekm-sumo) [inline deploy/helm/sumologic/templates/opentelemetry-operator/job-instrumentation-cr.yaml:10]: Why is this necessary?

File: deploy/helm/sumologic/values.yaml (hunk 1 above)
Comment (mat-rumian) [reply deploy/helm/sumologic/values.yaml:4548]: Yes, we observed on few deployments with opentelemetry-operator enabled that manager pod is crashing becuase of too low resources.

File: deploy/helm/sumologic/templates/opentelemetry-operator/job-instrumentation-cr.yaml (hunk 2 above)
Comment (mat-rumian) [reply deploy/helm/sumologic/templates/opentelemetry-operator/job-instrumentation-cr.yaml:10]: It's necessary for a case when customer want's to add additional namespace to be instrumented. For `helm upgrade...` existing job will be not triggered. There're two ways to solve this:
* delete job - I don't want to do that, in case of issues we will lose information about failure reason
* create new job with different name - it makes sure that job will be triggered again and operator will take care of creation of instrumentation resource

Comment (swiatekm-sumo) [reply deploy/helm/sumologic/templates/opentelemetry-operator/job-instrumentation-cr.yaml:10]: > For helm upgrade... existing job will be not triggered. 

Our Chart setup Job does get triggered every time. So why not this one?

Comment (sumo-drosiek) [reply deploy/helm/sumologic/templates/opentelemetry-operator/job-instrumentation-cr.yaml:10]: > > For helm upgrade... existing job will be not triggered.
> 
> Our Chart setup Job does get triggered every time. So why not this one?

Because we remove setup job when it finish

Comment (swiatekm-sumo) [reply deploy/helm/sumologic/templates/opentelemetry-operator/job-instrumentation-cr.yaml:10]: Can we do the same thing here?

File: deploy/helm/sumologic/Chart.yaml [hunk 3]
```diff
@@ -37,6 +37,6 @@ dependencies:
     repository: https://sumologic.github.io/tailing-sidecar
     condition: tailing-sidecar-operator.enabled
   - name: opentelemetry-operator
-    version: 0.18.3
+    version: 0.24.0
```
Comment (kkujawa-sumo) [inline deploy/helm/sumologic/Chart.yaml:40]: please update Opentelemtery Operator version in support matrix here: https://github.com/SumoLogic/sumologic-kubernetes-collection/tree/main/docs#support-matrix

File: deploy/helm/sumologic/templates/opentelemetry-operator/job-instrumentation-cr.yaml (hunk 2 above)
Comment (mat-rumian) [reply deploy/helm/sumologic/templates/opentelemetry-operator/job-instrumentation-cr.yaml:10]: We are going to lose information in case of failed instrumentation resource creation. It will make harder for us to investigate/troubleshoot issues.

Comment (swiatekm-sumo) [reply deploy/helm/sumologic/templates/opentelemetry-operator/job-instrumentation-cr.yaml:10]: Can't we just use the default hook delete policy from Helm: https://helm.sh/docs/topics/charts_hooks/#hook-deletion-policies? Then it's only going to be deleted after you try installing again, but that sounds like it should work if you want logs from failed executions.

The reason I'm asking about this is that adding the hash to the name feels like an antipattern in K8s and I'd rather avoid it.

Comment (sumo-drosiek) [reply deploy/helm/sumologic/templates/opentelemetry-operator/job-instrumentation-cr.yaml:10]: Hmm. Will that work for non-helm installation?

Comment (swiatekm-sumo) [reply deploy/helm/sumologic/templates/opentelemetry-operator/job-instrumentation-cr.yaml:10]: Does this work for setup in the non-Helm installation? If so, how?

Comment (sumo-drosiek) [reply deploy/helm/sumologic/templates/opentelemetry-operator/job-instrumentation-cr.yaml:10]: > Does this work for setup in the non-Helm installation

Good question. I don't know the answer :fearful:

**Review by swiatekm-sumo — APPROVED** [review]
